lib/validator.js

```diff
@@ -25,6 +25,15 @@ class Validator {
         return date instanceof Date
       }
     })
+
+    this.ajv.addKeyword({
+      keyword: 'isSet',
+      type: 'object',
+      errors: false,
+      validate: (type, data) => {
+        return data instanceof Set
```

Comment:
I didn't check the whole PR yet. But I think it's not enough here. You have to check that all restrictions like `items`, `minItems`, `maxItems` will work in this case.